glob.go

```diff
@@ -31,19 +31,24 @@ func Glob(pattern, subj string) bool {
 	end := len(parts) - 1
 
 	// Check the first section. Requires special handling.
-	if !leadingGlob && !strings.HasPrefix(subj, parts[0]) {
-		return false
+	if !leadingGlob {
+		if strings.HasPrefix(subj, parts[0]) {
+			// Strip prefix, to avoid matching it again
+			subj = subj[len(parts[0]):]
+		} else {
+			return false
+		}
 	}
 
 	// Go over the middle parts and ensure they match.
 	for i := 1; i < end; i++ {
-		if !strings.Contains(subj, parts[i]) {
+		partStartIdx := strings.Index(subj, parts[i])
+		if partStartIdx < 0 {
 			return false
 		}
 
 		// Trim evaluated text from subj as we loop over the pattern.
-		idx := strings.Index(subj, parts[i]) + len(parts[i])
-		subj = subj[idx:]
+		subj = subj[partStartIdx+len(parts[i]):]
```

Comment:
I think we can collapse this even further so that we only have one place where we trim the subject. I am thinking that a switch statement on `i` would work well for this case, that way we can just always trim the subject when a section matches. Let me know if that is clear or if you would like a sample of what I am talking about.